code-dot-org/code-dot-org
# Create the first version of the Oceans PWA

Create the first version of the Oceans PWA, which can only work with unoptimized assets.

[Demo slide deck 10/29/2020](https://docs.google.com/presentation/d/1G9z-m3uHyU9b9Osbjn8oj_J_OpIu-O19hKq4LATq9cM/edit#slide=id.p)

## Test
1. Start local server by running `bin/dashboard-server`
2. Go to `http://127.0.0.1:3000/pwa_demo`. 
3. In Chrome developer tool, go to Application tab, then check Manifest, Service workers and Cache Storage section. Verify that a page manifest is downloaded, a service worker is installed and the browser's Cache storage is updated with an application shell.  
  - <img width="957" alt="Screen Shot 2021-02-05 at 11 27 02 AM" src="https://user-images.githubusercontent.com/46507039/107080821-a5deb480-67a6-11eb-9cd1-b4ac47398ff1.png">

4. Click "Make Oceans tutorial available offline", wait until all assets are downloaded and a popup appears. 
  - <img width="707" alt="Screen Shot 2021-02-05 at 11 28 17 AM" src="https://user-images.githubusercontent.com/46507039/107080832-a7a87800-67a6-11eb-9fdb-6ebcf1d520e8.png">

5. Switch to offline mode.
  - <img width="957" alt="Screen Shot 2021-02-05 at 11 31 33 AM" src="https://user-images.githubusercontent.com/46507039/107080834-a8410e80-67a6-11eb-902c-efcc9362d8c9.png">

6. Verify all puzzle links work. (In puzzle 1, 5, 7, you will just see black boxes instead of videos because videos are not cached). Click "Back to the Demo page" to go back to the main demo page.

7. Install local PWA app. You can still use Chrome developer tool with the PWA app to simulate offline scenario.
  - <img width="534" alt="Screen Shot 2021-02-05 at 11 48 50 AM" src="https://user-images.githubusercontent.com/46507039/107082051-7cbf2380-67a8-11eb-98c9-51dd7aa73027.png">
  - <img width="440" alt="Screen Shot 2021-02-05 at 11 50 56 AM" src="https://user-images.githubusercontent.com/46507039/107082058-7e88e700-67a8-11eb-90c7-d4e57f3efccf.png">
  - <img width="1100" alt="Screen Shot 2021-02-05 at 11 50 06 AM" src="https://user-images.githubusercontent.com/46507039/107082053-7d57ba00-67a8-11eb-9683-e45b54f3c806.png">


## Links
Earlier work: https://github.com/code-dot-org/code-dot-org/compare/ha/oceans-cache
